pkg/vectorize/date_format/dateformat.go

```diff
@@ -0,0 +1,232 @@
+		}
+		buf.WriteString(MonthNames[m-1][:3])
+	case 'M':
+		m := t.Month()
+		if m == 0 || m > 12 {
+			return errors.New(errno.InvalidOptionValue, "the input paraemter value is in wrong format")
+		}
+		buf.WriteString(MonthNames[m-1])
+	case 'm':
+		buf.WriteString(FormatIntByWidth(int(t.Month()), 2))
+	case 'c':
+		buf.WriteString(strconv.FormatInt(int64(t.Month()), 10))
+	case 'D':
+		buf.WriteString(strconv.FormatInt(int64(t.Day()), 10))
+		buf.WriteString(AbbrDayOfMonth(int(t.Day())))
+	case 'd':
+		buf.WriteString(FormatIntByWidth(int(t.Day()), 2))
+	case 'e':
+		buf.WriteString(strconv.FormatInt(int64(t.Day()), 10))
+	case 'f':
+		fmt.Fprintf(buf, "%06d", t.MicroSec())
+	case 'j':
+		fmt.Fprintf(buf, "%03d", t.DayOfYear())
+	case 'H':
+		buf.WriteString(FormatIntByWidth(int(t.Hour()), 2))
+	case 'k':
+		buf.WriteString(strconv.FormatInt(int64(t.Hour()), 10))
+	case 'h', 'I':
+		tt := t.Hour()
+		if tt%12 == 0 {
+			buf.WriteString("12")
+		} else {
+			buf.WriteString(FormatIntByWidth(int(tt%12), 2))
+		}
+	case 'i':
+		buf.WriteString(FormatIntByWidth(int(t.Minute()), 2))
+	case 'l':
+		tt := t.Hour()
+		if tt%12 == 0 {
+			buf.WriteString("12")
+		} else {
+			buf.WriteString(strconv.FormatInt(int64(tt%12), 10))
+		}
+	case 'p':
+		hour := t.Hour()
+		if hour/12%2 == 0 {
+			buf.WriteString("AM")
+		} else {
+			buf.WriteString("PM")
+		}
+	case 'r':
+		h := t.Hour()
+		h %= 24
+		switch {
+		case h == 0:
+			fmt.Fprintf(buf, "%02d:%02d:%02d AM", 12, t.Minute(), t.Sec())
+		case h == 12:
+			fmt.Fprintf(buf, "%02d:%02d:%02d PM", 12, t.Minute(), t.Sec())
+		case h < 12:
+			fmt.Fprintf(buf, "%02d:%02d:%02d AM", h, t.Minute(), t.Sec())
+		default:
+			fmt.Fprintf(buf, "%02d:%02d:%02d PM", h-12, t.Minute(), t.Sec())
+		}
+	case 'S', 's':
+		buf.WriteString(FormatIntByWidth(int(t.Sec()), 2))
+	case 'T':
+		fmt.Fprintf(buf, "%02d:%02d:%02d", t.Hour(), t.Minute(), t.Sec())
+	case 'U':
+		w := t.Week(0)
+		buf.WriteString(FormatIntByWidth(w, 2))
+	case 'u':
+		w := t.Week(1)
+		buf.WriteString(FormatIntByWidth(w, 2))
+	case 'V':
+		w := t.Week(2)
+		buf.WriteString(FormatIntByWidth(w, 2))
+	case 'v':
+		_, w := t.YearWeek(3)
+		buf.WriteString(FormatIntByWidth(w, 2))
+	case 'a':
+		weekday := t.DayOfWeek()
+		buf.WriteString(AbbrevWeekdayName[weekday])
+	case 'W':
+		buf.WriteString(t.DayOfWeek().String())
+	case 'w':
+		buf.WriteString(strconv.FormatInt(int64(t.DayOfWeek()), 10))
+	case 'X':
+		year, _ := t.YearWeek(2)
+		if year < 0 {
+			buf.WriteString(strconv.FormatUint(uint64(math.MaxUint32), 10))
+		} else {
+			buf.WriteString(FormatIntByWidth(year, 4))
+		}
+	case 'x':
+		year, _ := t.YearWeek(3)
+		if year < 0 {
+			buf.WriteString(strconv.FormatUint(uint64(math.MaxUint32), 10))
+		} else {
+			buf.WriteString(FormatIntByWidth(year, 4))
+		}
+	case 'Y':
+		buf.WriteString(FormatIntByWidth(int(t.Year()), 4))
+	case 'y':
+		str := FormatIntByWidth(int(t.Year()), 4)
+		buf.WriteString(str[2:])
+	default:
```

Comment:
i think this case will:

case '%':
      buf.WriteRune(b)
default: 
     return throw some error

Reply:
The date_format() function handles the format string by processing the format matching symbol. If it is not the format matching symbol, it will be output as it is, such as:
mysql> select date_format('2022-03-21 12:34:12', '%Y-%m-%dsdfsf123456') from dual;
+-----------------------------------------------------------+
| date_format('2022-03-21 12:34:12', '%Y-%m-%dsdfsf123456') |
+-----------------------------------------------------------+
| 2022-03-21sdfsf123456                                     |
+-----------------------------------------------------------+